remove_association.hpp

```diff
@@ -0,0 +1,142 @@
+     *
+     * @param[in] bus - The sdbusplus bus object for making D-Bus calls
+     * @param[in] path - The root of the tree to search.
+     * @param[in] interface - Interface in the subtree to search for
+     * @param[in] depth - The number of path elements to descend.
+     */
+    virtual DBusSubtree listInventoryAssociations(sdbusplus::bus::bus& bus,
+                                                  const std::string& path,
+                                                  const std::string& interface,
+                                                  int32_t depth) const = 0;
+
+    /**
+     * @brief Sets the critical association on the D-Bus object.
+     *
+     * @param[in] objectPath - The D-Bus object path
+     */
+    virtual void
+        removeCriticalAssociation(const std::string& objectPath) const = 0;
+};
+
+/**
+ * @class RemoveAssociation
+ *
+ * Concrete implementation of RemoveAssociationBase.
+ */
+class RemoveAssociation : public RemoveAssociationBase
+{
+  public:
+    RemoveAssociation() = delete;
+    ~RemoveAssociation() = default;
+    RemoveAssociation(const RemoveAssociation&) = default;
+    RemoveAssociation& operator=(const RemoveAssociation&) = default;
+    RemoveAssociation(RemoveAssociation&&) = default;
+    RemoveAssociation& operator=(RemoveAssociation&&) = default;
+
+    /**
+     * @brief Constructor
+     *
+     * @param[in] bus - The sdbusplus bus object
+     */
+    explicit RemoveAssociation(sdbusplus::bus::bus& bus);
+
+    /**
+     * @brief Finds the D-Bus service name that hosts the
+     * passed in path and interface.
+     *
+     * @param[in] objectPath - The D-Bus object path
+     * @param[in] interface - The D-Bus interface
+     */
+    DBusService getService(const std::string& objectPath,
+                           const std::string& interface) const;
+
+    /**
+     * @brief Wrapper for the 'Get' properties method call
+     *
+     * @param[in] service - The D-Bus service to call it on
+     * @param[in] objectPath - The D-Bus object path
+     * @param[in] interface - The interface to get the property on
+     * @param[in] property - The property name
+     * @param[out] value - Filled in with the property value.
+     */
+    void getProperty(const std::string& service, const std::string& objectPath,
+                     const std::string& interface, const std::string& property,
+                     DBusValue& value) const;
+
+    /**
+     * @brief Get subtree from the object mapper
+     *
+     * @param[in] bus - The sdbusplus bus object for making D-Bus calls
+     * @param[in] path - The root of the tree to search.
+     * @param[in] interface - Interface in the subtree to search for
+     * @param[in] depth - The number of path elements to descend.
+     */
+    DBusSubtree listInventoryAssociations(sdbusplus::bus::bus& bus,
+                                          const std::string& path,
+                                          const std::string& interface,
```

Comment:
Since the name of the function already has 'Associations' in it, you could probably just hardcode the associations interface inside the function instead of passing it in if you so desired.

Reply:
Ack